src/api/app/models/staging/stage_requests.rb

```diff
@@ -48,7 +48,9 @@ def requests
   end
 
   def branch_package(bs_request_action)
+    staging_project = Staging::StagingProject.find_by(name: staging_project_name)
```

Comment:
Why don't you pass in the `StagingProject`? Instead of passing in the name (that you got from the project) and then reloading the project again?

Reply:
I also came across this. See #6354

Reply:
Because I have better things to do in my freetime:wink: